### PR TITLE
[tcat][tcat_ble_client] add TCAT Commissioner / Device certs  for Thread certification testing

### PR DESCRIPTION
**Pull request by EskoDijk**
Also with the BBTC Commissioner tool, adds scripts for example certificate generation; documentation updated to reflect this.
In [tcat_ble_client] it removse hostname use in BBTC - use of hostnames is not specified for TCAT. 
In [src/tcat] it fixes Thread-specific X509v3 extension parsing; unit test extended for this. 

The purpose of this change is to allow use of the BBTC client directly in TCAT cert testing. This requires X509 certificate identities that are based on the Thread Group internal test CA. The included private keys can be exposed - all for testing purposes.

The present generated certificates still use SKI/AKI extensions - this may be removed possibly later on in another PR, if analysis shows that these are not needed/required for TCAT use cases.

This PR partly addresses the issue #10196 - it at least prints a warning message when the TCAT Commissioner cert had an error in processing, and it attempts to close the TLS connection.  (To be verified later on if the closing actually works in the sense that the Commissioner receives the TLS alert / close-notify.) This avoids at least that during manual testing the connection looks "ok" but was silently rejected by the TCAT Device.

No CLANG/"Pretty" has been applied yet to the new code.

### CODE REVIEW

**Comment by EskoDijk** [comment]
FYI @canisLupus1313 with this PR, I can run TCAT in simulation using the Thread certs as defined by the test plan.

File: src/core/radio/ble_secure.cpp [hunk 1]
```diff
@@ -330,7 +330,12 @@ void BleSecure::HandleTlsConnected(bool aConnected)
 
         if (mTcatAgent.IsEnabled())
         {
-            IgnoreReturnValue(mTcatAgent.Connected(mTls));
+            Error err = mTcatAgent.Connected(mTls);
+            if (err != kErrorNone){
+                mTls.Close()
```
Comment (canisLupus1313) [inline src/core/radio/ble_secure.cpp:335]: Semicolon is missing here.

File: tools/tcat_ble_client/auth-generate/create-cert-ca.sh [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+openssl x509 -set_serial 0x01 -extfile "ext/${NAME}.ext" \
+ -extensions "${NAME}" -req -in "${NAME}.csr" \
+ -signkey "ca/${NAME}_key.pem" -out "ca/${NAME}_cert.pem" \
+ -days "${VALIDITY}" -sha256
+
+# delete temp files
+rm -f "${NAME}.csr"
+
+# show result
+openssl x509 -text -in "ca/${NAME}_cert.pem"
```
Comment (canisLupus1313) [inline tools/tcat_ble_client/auth-generate/create-cert-ca.sh:60]: Missing new line.

File: tools/tcat_ble_client/auth-generate/create-cert-tcat-commissioner.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  -CAkey "ca/${CANAME}_key.pem" -extfile "ext/${NAME}.ext" -extensions \
+  "${NAME}" -req -in "${NAME}.csr" -out "output/${NAME}/commissioner_cert.pem" \
+  -days "${VALIDITY}" -sha256
+
+# delete temp files
+rm -f "${NAME}.csr"
+
+# copy supporting files, for immediate use by TCAT Commissioner as 'cert_path'
+cp "${CACERTFILE}" "output/${NAME}/ca_cert.pem"
+cp "keys/${NAME}_key.pem" "output/${NAME}/commissioner_key.pem"
```
Comment (canisLupus1313) [inline tools/tcat_ble_client/auth-generate/create-cert-tcat-commissioner.sh:67]: Missing new line.

File: tools/tcat_ble_client/auth-generate/create-cert-tcat-device.sh [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+# delete temp files
+rm -f "${NAME}.csr"
+
+# copy supporting files, for immediate use by TCAT Commissioner as 'cert_path'.
+# Normally a Commissioner must not use Device certs, but for testing purposes this
+# option is provided here.
+cp "output/${NAME}/device_cert.pem" "output/${NAME}/commissioner_cert.pem"
+cp "${CACERTFILE}" "output/${NAME}/ca_cert.pem"
+cp "keys/${NAME}_key.pem" "output/${NAME}/commissioner_key.pem"
+cp "keys/${NAME}_key.pem" "output/${NAME}/device_key.pem"
```
Comment (canisLupus1313) [inline tools/tcat_ble_client/auth-generate/create-cert-tcat-device.sh:76]: Missing new line.

**Comment by canisLupus1313** [comment]
@EskoDijk  Overall looks good.

**Review by abtink — APPROVED** [review]
LGTM. Thanks.
Some smaller style suggestions below.

**Comment by EskoDijk** [comment]
@abtink @canisLupus1313 Thanks! I've updated with your suggestions and applied `make-pretty`.

**Review by canisLupus1313 — APPROVED** [review]

**Review by jwhui — APPROVED** [review]
Thanks! 👍🏼

**Pull request merged**